packages/core/src/command/hl7v2-subscriptions/hl7v2-to-fhir-converter.ts

```diff
@@ -0,0 +1,324 @@
+    newBundleUploadPromise,
+    combinedBundleUploadPromise,
+    bundlePresignedUrlPromise,
+  ]);
+
+  if (shouldNotifyApi && bundlePresignedUrl) {
+    try {
+      await executeWithNetworkRetries(
+        async () =>
+          await axios.post(internalHl7RouteUrl, undefined, {
+            params: {
+              cxId,
+              patientId,
+              presignedUrl: bundlePresignedUrl,
+            },
+          })
+      );
+      log(`Successfully sent HL7 FHIR bundle to the API`);
+    } catch (err) {
+      log(`Error hitting the API endpoint: - ${errorToString(err)}`);
+      throw err;
+    }
+  } else {
+    log(`Skipping API notification for message type: ${msgType.triggerEvent}`);
+  }
+}
+
+async function saturateConvertedBundle({
+  bundle,
+  patientId,
+  cxId,
+  messageId,
+  messageReceivedTimestamp,
+  internalGetPatientUrl,
+  log,
+}: {
+  bundle: Bundle<Resource>;
+  patientId: string;
+  cxId: string;
+  messageId: string;
+  messageReceivedTimestamp: string;
+  internalGetPatientUrl: string;
+  log: typeof console.log;
+}) {
+  try {
+    const patient = await executeWithNetworkRetries(
+      async () => await axios.get(internalGetPatientUrl)
+    );
+    const fhirPatient = toFhirPatient({ id: patientId, data: patient.data });
+    const fhirPatientEntry = buildBundleEntry(fhirPatient);
+    const combinedEntries = bundle.entry ? [fhirPatientEntry, ...bundle.entry] : [];
+    return buildBundle({ type: "collection", entries: combinedEntries });
+  } catch (error) {
+    const msg = "Error retrieving patient data on HL7-to-FHIR conversion";
+    log(`${msg}: ${errorToString(error)}`);
+    capture.message(msg, {
+      extra: {
+        patientId,
+        cxId,
+        messageId,
+        timestamp: messageReceivedTimestamp,
+      },
+    });
+
+    throw error;
+  }
+}
+
+async function mergeIncomingBundleIntoCombined({
+  cxId,
+  patientId,
+  existingBundle,
+  incomingBundle,
+  log,
+}: {
+  cxId: string;
+  patientId: string;
+  existingBundle: Bundle<Resource>;
+  incomingBundle: Bundle<Resource>;
+  log: typeof console.log;
+}) {
+  try {
+    const combinedEntries = [...(existingBundle.entry || []), ...(incomingBundle.entry || [])];
+    const combinedBundle = buildBundle({
+      type: "collection",
+      entries: combinedEntries,
+    });
+
+    const dedupedCombinedBundle = await deduplicate({
+      cxId,
+      patientId,
+      bundle: combinedBundle,
+    });
+    log(`Combined and deduped the incoming bundle with the existing one.`);
+
+    return dedupedCombinedBundle;
+  } catch (error) {
+    log(`Error during bundle merging or deduplication: ${errorToString(error)}`);
+    capture.error("Failed to merge or deduplicate bundles", {
+      extra: {
+        patientId,
+        cxId,
+        error,
+        entriesCount: {
+          existing: existingBundle.entry?.length || 0,
+          incoming: incomingBundle.entry?.length || 0,
+        },
+      },
```

Comment:
In this case if we want the entriesCount, we can Sentry.setExtra(entriesCount, { existing: ..., incoming: ... }) in the function while avoiding the branching of a try catch